lib/cast/cast_sender.js

```diff
@@ -583,6 +574,35 @@ shaka.cast.CastSender = class {
   }
 
 
+  /**
+   * @param {boolean} available
+   * @private
+   */
+  onGCastApiAvailable_(available) {
+    // restore callback from saved
+    if (this.__onGCastApiAvailable_) {
+      window.__onGCastApiAvailable = this.__onGCastApiAvailable_;
+    } else {
+      delete window.__onGCastApiAvailable;
+    }
+    this.__onGCastApiAvailable_ = null;
+
+    // A note about this value: In our testing environment, we load both
+    // uncompiled and compiled code.  This global callback in uncompiled mode
+    // can be overwritten by the same in compiled mode.  The two versions will
+    // each have their own instances_ map.  Therefore the callback must have a
+    // name, as opposed to being anonymous.  This way, the CastSender tests
+    // can invoke the named static method instead of using a global that could
+    // be overwritten.
+    shaka.cast.CastSender.onSdkLoaded_(available);
+
+    // call restored callback (if any)
+    if (typeof window.__onGCastApiAvailable === 'function') {
```

Comment:
If the user makes multiple cast senders, wouldn't this "fall through" behavior lead to `onGCastApiAvailable_` being called multiple times before falling through to the pre-existing `__onGCastApiAvailable` callback? That would then result in each cast sender being re-initialized multiple times.

I think it would be better if this `shaka.cast.CastSender.onGCastApiAvailable_` method was static, and the `__onGCastApiAvailable_` member variable was stored globally as well. That way, we could check to see if the `__onGCastApiAvailable` callback is already set to `shaka.cast.CastSender.onGCastApiAvailable_`, and not replace it if it already is.

Reply:
Good point!

Reply:
Agreed. Although the corner case of another party registering a callback that wraps the shaka callback (thus defeating the check) makes this a bit tricky. Will work something out.

Reply:
Implemented it as suggested. The above corner case is rather theoretic, and will work correctly too, albeit less optimally.